examples/raw-template/raw_template.bal

```diff
@@ -0,0 +1,63 @@
+import ballerina/lang.'object;
+import ballerina/io;
+
+public function main() {
+    string name = "Ballerina";
+    // Create a raw template with the `name` variable interpolated in the
+    // literal.
+    'object:RawTemplate template = `Hello ${name}!!!`;
+    // The string literal portions of the raw template are accessible
+    // through the `strings` field. The `strings` field is a list of
+    // `string` values. The string literal portions of the raw template
+    // are separated by the interpolations. If there are `n` insertions,
+    // there will be `n+1` strings.
+    io:println(template.strings);
+    // The `insertions` field holds the values obtained by evaluating
+    // the expressions specified in the interpolations of the raw template.
+    // In this case, there is just one interpolation: a reference to the
+    // variable, `name`.
+    io:println(template.insertions[0]);
+
+    Person[] people = [
+        {name: "John Doe", age: 25},
+        {name: "Jane Doe", age: 20}
+    ];
+
+    Query[] queries = [];
+    int i = 0;
+
+    foreach var p in people {
+        queries[i] = (`INSERT INTO Details VALUES (${p.name}, ${p.age})`);
+        i += 1;
+    }
+
+    io:println(buildQuery(queries[0]));
+    io:println(buildQuery(queries[1]));
+}
+
+// A user can define their own raw template types as well. To define a
+// raw template type,<br>
+// 1) The type should be an abstract subtype of `object:RawTemplate` type<br>
+// 2) It should only have the two fields `strings` and `insertions`<br>
+// 3) It should not have method declarations
+type Query abstract object {
+    public string[] strings;
+    // The type of the fields can be used to constrain the raw template
+    // in ways such as limiting the number of interpolations/strings
+    // that can be specified, ensuring that the interpolations are of
+    // correct types etc.
+    public [string, int] insertions;
+};
```

Comment:
Shouldn't this comment come before `strings`?

```suggestion
    // The types of the fields can be used to enforce constraints 
    // on the raw template.
    // For example, a fixed-length list can limit the number of 
    // strings and insertions.
    public string[3] strings;
    // The type of `insertions` can enforce the insertions to
    // be of particular types.
    public [string, int] insertions;
};
```

Reply:
But wouldn't it then show the comment against the string field? Put it for the insertions field since it's a different type from the type defined in `RawTemplate`

Reply:
But it's common to both right? Using a fixed length list?

What I thought was, if we have it for `strings`, when following the flow, when we get to insertions we still know it to apply it for insertions too. 

In the current approach though we are saying something applies for `strings` too after the definition of `strings`.